streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -570,167 +636,184 @@ public void shouldFailWithMigrateExceptionOnEOSProcessUnknownPid() {
         props.setProperty(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, StreamsConfig.EXACTLY_ONCE);
         final RecordCollector collector = new RecordCollectorImpl(
             taskId,
             new StreamsConfig(props),
             logContext,
             streamsMetrics,
-            null,
+            consumer,
             id -> new MockProducer<byte[], byte[]>(cluster, true, new DefaultPartitioner(), byteArraySerializer, byteArraySerializer) {
                 @Override
-                public void sendOffsetsToTransaction(final Map<TopicPartition, OffsetAndMetadata> offsets, final String consumerGroupId) {
-                    throw new KafkaException("boom");
+                public void abortTransaction() {
+                    functionCalled.set(true);
+                    super.abortTransaction();
                 }
             }
         );
 
-        assertThrows(StreamsException.class, () -> collector.commit(Collections.singletonMap(new TopicPartition("topic", 0), new OffsetAndMetadata(5L))));
+        collector.send(topic, "3", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
+        collector.commit(Collections.emptyMap());
+
+        collector.close();
+
+        assertFalse(functionCalled.get());
     }
 
     @Test
-    public void shouldFailWithMigrateExceptionOnEOSCommitUnexpected() {
+    public void shouldThrowStreamsExceptionOnEOSAbortTxnFatalException() {
+        final KafkaException exception = new KafkaException();
         final Properties props = StreamsTestUtils.getStreamsConfig("test");
         props.setProperty(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, StreamsConfig.EXACTLY_ONCE);
         final RecordCollector collector = new RecordCollectorImpl(
             taskId,
             new StreamsConfig(props),
             logContext,
             streamsMetrics,
-            null,
+            consumer,
             id -> new MockProducer<byte[], byte[]>(cluster, true, new DefaultPartitioner(), byteArraySerializer, byteArraySerializer) {
                 @Override
-                public void commitTransaction() {
-                    throw new KafkaException("boom");
+                public void abortTransaction() {
+                    throw exception;
                 }
             }
         );
 
-        assertThrows(StreamsException.class, () -> collector.commit(Collections.singletonMap(new TopicPartition("topic", 0), new OffsetAndMetadata(5L))));
+        collector.send(topic, "3", "0", null, null, stringSerializer, stringSerializer, streamPartitioner);
+
+        final StreamsException thrown = assertThrows(StreamsException.class, collector::close);
+        assertEquals(exception, thrown.getCause());
     }
 
-    @Test(expected = StreamsException.class)
-    public void shouldThrowIfTopicIsUnknownWithDefaultExceptionHandler() {
+    @Test
+    public void shouldSwallowOnEOSAbortTxnFatalException() {
```

Comment:
q: what does this test cover? The `abortTxn` call is not triggered in this case right?

Reply:
Good catch! I will change the test code.